shims/spark34/src/main/scala/io/glutenproject/sql/shims/spark34/Spark34Shims.scala

```diff
@@ -252,4 +254,45 @@ class Spark34Shims extends SparkShims {
   def getAnalysisExceptionPlan(ae: AnalysisException): Option[LogicalPlan] = {
     ae.plan
   }
+
+  override def getKeyGroupedPartitioning(batchScan: BatchScanExec): Option[Seq[Expression]] = {
+    batchScan.keyGroupedPartitioning
+  }
+
+  override def getCommonPartitionValues(
+      batchScan: BatchScanExec): Option[Seq[(InternalRow, Int)]] = {
+    batchScan.commonPartitionValues
+  }
+
+  override def orderPartitions(
```

Comment:
Shouldn't  have the partitions distribution been done at plan analysis phase? Why do we need to add this step at transformer layer?

Reply:
The partitions are ordered when partitions are accessed in wholestagecodegen, since we override wholestagecodegen we need to handle in transformer layer.